server/src/main/java/io/crate/execution/ddl/tables/TableCreator.java

```diff
@@ -51,24 +59,46 @@ public TableCreator(TransportCreateTableAction transportCreateIndexAction) {
         this.transportCreateTableAction = transportCreateIndexAction;
     }
 
-    public CompletableFuture<Long> create(BoundCreateTable createTable) {
+    public CompletableFuture<Long> create(BoundCreateTable createTable, Version minNodeVersion) {
         var templateName = createTable.templateName();
         var relationName = createTable.tableIdent();
-        var createTableRequest = templateName == null
-            ? new CreateTableRequest(
-                new CreateIndexRequest(
-                    relationName.indexNameOrAlias(),
-                    createTable.tableParameter().settings()
-                ).mapping(createTable.mapping())
+        CreateTableRequest createTableRequest;
+
+        if (minNodeVersion.onOrAfter(Version.V_5_4_0)) {
+            LinkedHashMap<ColumnIdent, Reference> references = new LinkedHashMap<>();
+            IntArrayList pKeysIndices = new IntArrayList();
+            createTable.analyzedTableElements().collectReferences(relationName, references, pKeysIndices, true);
+            var policy = (String) createTable.tableParameter().mappings().get(ColumnPolicies.ES_MAPPING_NAME);
+            createTableRequest = new CreateTableRequest(
+                relationName,
+                new ArrayList<>(references.values()),
+                pKeysIndices,
+                createTable.analyzedTableElements().getCheckConstraints(),
+                createTable.tableParameter().settings(),
+                createTable.routingColumn(),
+                policy != null ? ColumnPolicies.decodeMappingValue(policy) : ColumnPolicy.STRICT,
+                createTable.partitionedBy(),
+                createTable.analyzedTableElements().indicesMap()
+            );
+
+        } else {
+            // TODO: Remove in 5.5.
+            createTableRequest = templateName == null
+                ? new CreateTableRequest(
+                    new CreateIndexRequest(
+                        relationName.indexNameOrAlias(),
+                        createTable.tableParameter().settings()
+                    ).mapping(createTable.mapping())
```

Comment:
We could, if we are fine with the fact that in the mixed cluster, some 5.3 node will get old-style requests with mapping generated with the new code. Should be fine?

Reply:
Afaik the mapping must stay BWC anyway, so it should be fine or am I missing something?

Reply:
Yes, that's true, it must stay BWC. I was thinking about worst case when new mapping building code has a bug but it doesn't change anything we will see it anyway. Will remove `BoundCreateTable.mapping `and consequently, `AnalyzedTableElements.toMapping` now.

Reply:
>  I was thinking about worst case when new mapping building code has a bug 
Exactly for that reason it would be even better if we use it everywhere and do not use 2 different code paths so that any bug will be visible right now.